Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.cpp

```diff
@@ -241,6 +246,79 @@ namespace ShaderManagementConsole
         // Reconnect to the notification bus now that all changes have been applied
         AtomToolsFramework::AtomToolsDocumentNotificationBus::Handler::BusConnect(m_toolId);
     }
+
+    void ShaderManagementConsoleTableView::OnOptionClicked(int column)
+    {
+        const auto& shaderOptionDescriptor = m_shaderOptionDescriptors[column];
+
+        AZStd::vector<AZ::Data::AssetId> materialAssetIdList;
+        ShaderManagementConsoleRequestBus::BroadcastResult(
+            materialAssetIdList,
+            &ShaderManagementConsoleRequestBus::Events::GetAllMaterialAssetIds);
+
+        QString materialList = "";
+
+        QProgressDialog progressDialog("Gather shader variant information...", "Cancel", 0, materialAssetIdList.size());
+        progressDialog.setWindowModality(Qt::WindowModal);
+        progressDialog.setMaximumWidth(400);
+        progressDialog.setMaximumHeight(100);
+        progressDialog.setWindowTitle("Gather information from material assets");
```

Comment:
Couple of suggestions here:
- Any user facing strings should be wrapped in `tr()` so they can be translated in the future
- The default cancel button text is "Cancel", it doesn't need to be specified
- Any modal dialogs should get parented to the currently active window instead of `nullptr` so they get positioned properly, which we have a nice helper method for
- The default minimum is already 0 so we don't need to specify it

```suggestion
        QProgressDialog progressDialog(AzToolsFramework::GetActiveWindow());
        progressDialog.setWindowModality(Qt::WindowModal);
        progressDialog.setMaximum(materialAssetIdList.size());
        progressDialog.setMaximumWidth(400);
        progressDialog.setMaximumHeight(100);
        progressDialog.setWindowTitle(tr("Gather information from material assets"));
        progressDialog.setLabelText(tr("Gather shader variant information..."));
```